x-pack/plugins/stack_connectors/server/connector_types/bedrock/bedrock.ts

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ServiceParams, SubActionConnector } from '@kbn/actions-plugin/server';
+import aws from 'aws4';
+import type { AxiosError } from 'axios';
+import {
+  BedrockRunActionParamsSchema,
+  BedrockRunActionResponseSchema,
+  InvokeAIActionParamsSchema,
+} from '../../../common/bedrock/schema';
+import type {
+  BedrockConfig,
+  BedrockSecrets,
+  BedrockRunActionParams,
+  BedrockRunActionResponse,
+  InvokeAIActionParams,
+  InvokeAIActionResponse,
+} from '../../../common/bedrock/types';
+import { SUB_ACTION } from '../../../common/bedrock/constants';
+
+interface SignedRequest {
+  method: string;
+  service: string;
+  region: string;
+  host: string;
+  headers: Record<string, string>;
+  body: string;
+  path: string;
+}
+
+export class BedrockConnector extends SubActionConnector<BedrockConfig, BedrockSecrets> {
+  private url;
+  private model;
+  private region;
+
+  constructor(params: ServiceParams<BedrockConfig, BedrockSecrets>) {
+    super(params);
+
+    this.url = this.config.apiUrl;
+    this.model = this.config.defaultModel;
+    this.region = this.config.region;
+
+    this.registerSubActions();
+  }
+
+  private registerSubActions() {
+    this.registerSubAction({
+      name: SUB_ACTION.RUN,
+      method: 'runApi',
+      schema: BedrockRunActionParamsSchema,
+    });
+
+    this.registerSubAction({
+      name: SUB_ACTION.TEST,
+      method: 'runApi',
+      schema: BedrockRunActionParamsSchema,
+    });
+
+    this.registerSubAction({
+      name: SUB_ACTION.INVOKE_AI,
+      method: 'invokeAI',
+      schema: InvokeAIActionParamsSchema,
+    });
+  }
+
+  protected getResponseErrorMessage(error: AxiosError<{ error?: { message?: string } }>): string {
+    if (!error.response?.status) {
+      return `Unexpected API Error: ${error.code} - ${error.message}`;
+    }
+    if (error.response.status === 401) {
+      return 'Unauthorized API Error';
+    }
+    return `API Error: ${error.response?.status} - ${error.response?.statusText}${
+      error.response?.data?.error?.message ? ` - ${error.response.data.error?.message}` : ''
```

Comment:
When I use an invalid model, right now it just returns `Error: Status code: 400. Message: API Error: 400 - Bad Request` but there is a more informative error message inside `error.response.data.message` that would be useful `The provided model identifier is invalid.`

Reply:
Definitely, i kind of breezed over error handling because I have another ticket to clean it up in GenAI connector so I was going to handle all of that at once if that is ok? https://github.com/elastic/security-team/issues/7373

Reply:
Here is the preview of that PR: https://github.com/stephmilovic/kibana/pull/13


I will reopen against main once this PR + https://github.com/stephmilovic/kibana/pull/12 are merged